src/main/java/io/zipcoder/Student.java

```diff
@@ -15,7 +19,21 @@
      * @param lastName
      */
     public Student(String firstName, String lastName) {
+        this.firstName = firstName;
+        this.lastName = lastName;
+        this.testScores = new ArrayList<Double>();
+        this.examsTaken = 0;
+    }
 
+    public Student(String firstName, String lastName, Double[] testScores) {
+        this.firstName = firstName;
+        this.lastName = lastName;
+        this.testScores = new ArrayList<Double>();
+        for (Double score: testScores) {
+            this.testScores.add(score);
+        }
```

Comment:
you really don't need to do this. a Double[] can be a argument to arraylist constructor. check the ref docs.